### ISSUE TITLE
Scene Management (Timing?) Issue when starting other scenes

### ISSUE THREAD

**Issue by max1701** [photonstorm/phaser#3314]
Hi!

### Setup
I'm using Phaser 3.1.2 and in my very simple scenario I create 3 scenes, A, B and C. 

Game starts in scene A which loads a simple image in its preload function; in it's create function I simply `this.scene.start('B')`.

Scene B has no preload function and in it's create function I just invoke `this.scene.start('C')`.

Scene C has no preload function and in it's create function it just displays the image.

I assembled a [GIST](https://gist.github.com/max1701/d9052e13f364f996744b5141dc8f854a) to illustrate the scenario and to reproduce the issue.

### The problem
Running the GIST as-is,  no image is shown (and no error messages are logged in the console)!

As commented in the GIST, I found two ways to fix the issue.

To get a rendered image, I either had to provide a non-empty preload function in Scene B so, the create gets delayed somehow or, without a non-empty preload function, I had to wrap the `this.scene.start('C')` in a setTimeout with a delay of at least 1 ms.

Thanks,
and greetings from austria

**Comment by Rybar** [comment]
A race condition created by the image not being loaded yet. There's multiple ways to deal with this, but here's how I'd handle it: In Scene B's update function I'd check to see if the image key exists, and then load the next scene.

```if(this.textures.exists('imageFromSceneA')) this.scene.start('sceneC')```

**Comment by samme** [comment]
That sounds like a bug, since A.create shouldn't run before its load queue has completed.

**Comment by max1701** [comment]
I think, scene A is properly waiting for its load queue to complete, since if you bypass scene B completely by calling 

`this.scene.start('sceneC')` in Scene As create method everything works as expected and the image gets rendered.

So it seems the bug gets introduced by adding the 'empty'  scene B.

If it's a race condition as supposed by @Rybar, it's interesting that the race condition occurs in the pressence of scene B but does not occur if scene B is bypassed and scene A starts scene C directly.

**Comment by photonstorm** [comment]
I don't believe this is a race condition, as `create` is never called until `preload` has finished.

However, the gist given works fine for me. I extended it to this test which goes from Scene A through to E (with B, C and D all blank) and it works too, locally and from a server:

http://labs.phaser.io/view.html?src=src\scenes\change%20scene%20from%20create.js

This is with the latest build from the master branch.

**Comment by photonstorm** [comment]
Here's an ES6 version too:

http://labs.phaser.io/view.html?src=src\scenes\change%20scene%20from%20create%20es6.js

**Comment by rblopes** [comment]
Take off the `preload` step from SceneA, and we have this:

![spectacle ns4623](https://user-images.githubusercontent.com/61425/36977015-680c7332-205e-11e8-8e8b-d804b476da6b.png)

<details>
<summary>Sample code (modified from last example, tested in the Labs)</summary>

```js
class SceneA extends Phaser.Scene {

    constructor ()
    {
        super('sceneA');
    }

    // preload ()
    // {
    //     this.load.image('face', 'assets/pics/bw-face.png');
    //     this.load.image('arrow', 'assets/sprites/longarrow.png');
    // }

    create ()
    {
        /* Should yield: SceneB { ... } */
        console.log(this.scene.get('sceneB'));
        this.scene.start('sceneB');
    }

}

class SceneB extends Phaser.Scene {

    constructor ()
    {
        super('sceneB');
    }

    create ()
    {
        /* Should yield: SceneC { ... } */
        console.log(this.scene.get('sceneC'));
        this.scene.start('sceneC');
    }

}

class SceneC extends Phaser.Scene {

    constructor ()
    {
        super('sceneC');
    }

    create ()
    {
        /* Should yield: SceneD { ... } */
        console.log(this.scene.get('sceneD'));
        this.scene.start('sceneD');
    }

}

class SceneD extends Phaser.Scene {

    constructor ()
    {
        super('sceneD');
    }

    create ()
    {
        /* Should yield: SceneE { ... } */
        console.log(this.scene.get('sceneE'));
        this.scene.start('sceneE');
    }

}

class SceneE extends Phaser.Scene {

    constructor ()
    {
        super('sceneE');
    }

    create ()
    {
        // this.face = this.add.image(400, 300, 'face');
        // this.arrow = this.add.sprite(400, 300, 'arrow').setOrigin(0, 0.5);
    }

    update ()
    {
        // this.arrow.rotation += 0.01;
    }

}

const game = new Phaser.Game({
    type: Phaser.AUTO,
    width: 800,
    height: 600,
    parent: 'phaser-example',
    scene: [ SceneA, SceneB, SceneC, SceneD, SceneE ],

    /*  Alternative: add scenes after game is booted. */
    // callbacks: {
    //     postBoot() {
    //         for (const scene of [ SceneA, SceneB, SceneC, SceneD, SceneE ]) {
    //             game.scene.add(null, scene);
    //         }
    //         game.scene.start('sceneA');
    //     }
    // }
});
```
</details>

The other scenes are reported in the console (`SceneC { ... }`, etc), of couse, but this happens because the scenes have been properly added on the manager by that moment and `start` is scheduled rather than synchronous. Put back `SceneA#preload` again (or add scenes after the game is booted) and SceneB is reported as should.

Note that I've reported the same problem [before](https://github.com/photonstorm/phaser/issues/3185), but did not extend the samples to run with more scenes.

**Comment by photonstorm** [comment]
If I remove the `preload` from SceneA they all still boot in the correct order.

The issue here is the use of `getScene` during the boot sequence.

**Comment by photonstorm** [comment]
Ok, have refactored how the initial boot scenes are handled, so `getScene` and similar should work fine now, even when called in a massive `create` chain during game boot. All changes in the master branch.

**Comment by rblopes** [comment]
I was about to reply your post when you committed 7c16368d3e46467b95faec594b54e197d1827cf8.

Before that, I tested the Gist code the OP posted against 3.1.2 and the (then) master branch, concluding that the issue posted was fixed.

The reason I thought this issue was similar to mine is because he mentions delaying the scene initialization as a temporary fix.

I'll test the recent changes and save the discussion for my original post. Sorry for the trouble.
